aten/src/TH/generic/THTensorApply.hpp

```diff
@@ -160,7 +160,7 @@ if (std::isnan(val)) break;
 static inline scalar_t THTensor_(powOne)(scalar_t x, scalar_t y) {
 #if defined(TH_REAL_IS_FLOAT)
   return powf(x, y);
-#elif defined(TH_REAL_IS_DOUBLE)
+#elif defined(TH_REAL_IS_DOUBLE) || defined(TH_REAL_IS_BOOL)
```

Comment:
Im getting an error during the CI build: 
/var/lib/jenkins/workspace/aten/src/TH/generic/THTensorApply.hpp: In function 'bool THBoolTensor_powOne(bool, bool)':

/var/lib/jenkins/workspace/aten/src/TH/generic/THTensorApply.hpp:166:16: error: comparison of constant '0' with boolean expression is always true [-Werror=bool-compare]

/var/lib/jenkins/workspace/build/caffe2/aten/src/TH/THGeneral.h:116:35: note: in definition of macro 'THArgCheck'

Reply:
As far as i understand, THBoolTensor_powOne is being generated because aten/src/TH/generic/THTensorApply.hpp is being included in the THTensorEvenMoreMath.cpp

Solution to this issue depends on the decision about either _fill & _zero should be moved or not.